jspoon/src/test/java/pl/droidsonroids/jspoon/FieldTypeTest.java

```diff
@@ -0,0 +1,214 @@
+    public void testObjectList() throws Exception {
+        HtmlAdapter<WildcardModel> htmlAdapter = Jspoon.create().adapter(WildcardModel.class);
+        WildcardModel model = htmlAdapter.fromHtml(HTML_CONTENT);
+        assertNotNull(model.list);
+        assertFalse(model.list.isEmpty());
+        assertEquals("123", model.list.get(0));
+    }
+
+    static class BoundWildcardModel {
+        @Selector(".number") List<? extends String> list;
+    }
+
+    @Test
+    public void testStringList() throws Exception {
+        HtmlAdapter<BoundWildcardModel> htmlAdapter = Jspoon.create().adapter(BoundWildcardModel.class);
+        BoundWildcardModel model = htmlAdapter.fromHtml(HTML_CONTENT);
+        assertNotNull(model.list);
+        assertFalse(model.list.isEmpty());
+        assertEquals("123", model.list.get(0));
+    }
+
+    static class ModelGeneric<T> {
+        @SuppressWarnings("rawtypes")
+        @Selector("div > div") List justList;
+        @Selector("span") Collection<?> listAny;
+        @Selector(".float") Set<? extends T> listExtType;
+        @Selector(".negative") LinkedHashSet<? super T> listSupType;
+        @Selector(".positive") LinkedList<T> listType;
+    }
+
+    static class ModelString extends ModelGeneric<String> {
+    }
+    static class ModelBigDecimal extends ModelGeneric<BigDecimal> {
+    }
+
+    @Test
+    public void genericFieldsTest() throws Exception {
+        @SuppressWarnings("rawtypes")
+        HtmlAdapter<ModelGeneric> htmlAdapter = jspoon.adapter(ModelGeneric.class);
+        ModelGeneric<?> model = htmlAdapter.fromHtml(HTML_CONTENT);
+        assertListEquals(model.justList, "3366", "-4", "9.98");
+        assertListEquals(model.listAny, "123", "-200", "3.14");
+        assertListEquals(model.listExtType, "3.14", "9.98", "2.76");
+        assertListEquals(model.listSupType, "-200", "-4", "-1");
+        assertListEquals(model.listType, "123", "3366", "1314");
+    }
+
+    @Test
+    public void genericStringFieldsTest() throws Exception {
+        HtmlAdapter<ModelString> htmlAdapter = jspoon.adapter(ModelString.class);
+        ModelString model = htmlAdapter.fromHtml(HTML_CONTENT);
+        assertListEquals(model.justList, "3366", "-4", "9.98");
+        assertListEquals(model.listAny, "123", "-200", "3.14");
+        assertListEquals(model.listExtType, "3.14", "9.98", "2.76");
+        assertListEquals(model.listSupType, "-200", "-4", "-1");
+        assertListEquals(model.listType, "123", "3366", "1314");
+    }
+
+    @Test
+    public void genericBigDecimalFieldsTest() throws Exception {
+        HtmlAdapter<ModelBigDecimal> htmlAdapter = jspoon.adapter(ModelBigDecimal.class);
+        ModelBigDecimal model = htmlAdapter.fromHtml(HTML_CONTENT);
+        assertListEquals(model.justList, "3366", "-4", "9.98");
+        assertListEquals(model.listAny, "123", "-200", "3.14");
+        assertListEquals(model.listExtType, new BigDecimal("3.14"), new BigDecimal("9.98"), new BigDecimal("2.76"));
```

Comment:
This fails as `BigDecimal` should be decimal ;)
```assertListEquals(model.listExtType, new BigDecimal("3"), new BigDecimal("9"), new BigDecimal("2"));```

Reply:
And here I thought checking "Allow edits from maintainers." box would speed things up :)